pkg/apiserver/patch/patch.go

```diff
@@ -0,0 +1,454 @@
+/*
```

Comment:
Any reason this couldn't go into `pkg/util/patch/*`?  Feels more generic than apiserver.